aws/data_source_aws_ecr_repository.go

```diff
@@ -45,11 +45,6 @@ func dataSourceAwsEcrRepositoryRead(d *schema.ResourceData, meta interface{}) er
 	log.Printf("[DEBUG] Reading ECR repository: %s", params)
 	out, err := conn.DescribeRepositories(params)
 	if err != nil {
-		if isAWSErr(err, ecr.ErrCodeRepositoryNotFoundException, "") {
-			log.Printf("[WARN] ECR Repository %s not found, removing from state", d.Id())
-			d.SetId("")
-			return nil
-		}
```

Comment:
let's keep this `if` condition to check for this type of error as it's still applicable and helps a user read through the error messaging but instead of logging the message, setting the ID, and returning nil we can just return on error per #13410 e.g.
```go
if isAWSErr(err, ecr.ErrCodeRepositoryNotFoundException, "") {
    return fmt.Errorf("ECR Repository (%s) not found",  d.Get(name).(string))
}
```